lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1426,20 +1432,21 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     return SV;
   }
   case Intrinsic::ptr_annotation: {
-    GetElementPtrInst *GI;
-    if (auto *BI = dyn_cast<BitCastInst>(II->getArgOperand(0))) {
-      GI = dyn_cast<GetElementPtrInst>(BI->getOperand(0));
-    } else {
-      GI = dyn_cast<GetElementPtrInst>(II->getOperand(0));
-    }
-
     GetElementPtrInst *GEP = dyn_cast<GetElementPtrInst>(II->getArgOperand(1));
     Constant *C = dyn_cast<Constant>(GEP->getOperand(0));
     // TODO: Refactor to use getConstantStringInfo()
     StringRef AnnotationString =
         dyn_cast<ConstantDataArray>(C->getOperand(0))->getAsCString();
 
-    if (GI) {
+    // Strip all bitcast and addrspace casts from the pointer argument:
+    //   llvm annotation intrinsic only takes i8*, so the original pointer
+    //   probably had to loose its addrspace and its original type.
+    Value *AnnotSubj = II->getArgOperand(0);
+    while (isa<BitCastInst>(AnnotSubj) || isa<AddrSpaceCastInst>(AnnotSubj)) {
+      AnnotSubj = cast<CastInst>(AnnotSubj)->getOperand(0);
```

Comment:
What if we have cast of cast or cast of addrspacecast, i.e. more than one cast?

Reply:
It's not possible to happen. We will always have bitcast or addrspacecast.

Reply:
Oh, you actually have a `while` loop here which will unwrap nested casts if any. Please, disregard my comment then